sdk/signalr/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/core/rest/_rest_py3.py

```diff
@@ -0,0 +1,716 @@
+
+    def __repr__(self) -> str:
+        content_type_str = (
+            ", Content-Type: {}".format(self.content_type) if self.content_type else ""
+        )
+        return "<{}: {} {}{}>".format(
+            type(self).__name__, self.status_code, self.reason, content_type_str
+        )
+
+    def _validate_streaming_access(self) -> None:
+        if self.is_closed:
+            raise ResponseClosedError()
+        if self.is_stream_consumed:
+            raise StreamConsumedError()
+
+class HttpResponse(_HttpResponseBase):
+
+    @property
+    def content(self):
+        # type: (...) -> bytes
+        try:
+            return self._content
+        except AttributeError:
+            raise ResponseNotReadError()
+
+    def close(self) -> None:
+        self.is_closed = True
+        self._internal_response.internal_response.close()
+
+    def __exit__(self, *args) -> None:
+        self._internal_response.internal_response.__exit__(*args)
+
+    def read(self) -> bytes:
+        """
+        Read the response's bytes.
+
+        """
+        try:
+            return self._content
+        except AttributeError:
+            self._validate_streaming_access()
+            self._content = (
+                self._internal_response.body() or
+                b"".join(self.iter_raw())
+            )
+            self._close_stream()
+            return self._content
+
+    def iter_bytes(self, chunk_size: int = None) -> Iterator[bytes]:
+        """Iterate over the bytes in the response stream
+        """
+        try:
+            chunk_size = len(self._content) if chunk_size is None else chunk_size
+            for i in range(0, len(self._content), chunk_size):
+                yield self._content[i: i + chunk_size]
+
+        except AttributeError:
+            for raw_bytes in self.iter_raw(chunk_size=chunk_size):
+                yield raw_bytes
+
+    def iter_text(self, chunk_size: int = None) -> Iterator[str]:
+        """Iterate over the response text
+        """
+        for byte in self.iter_bytes(chunk_size):
+            text = byte.decode(self.encoding or "utf-8")
+            yield text
+
+    def iter_lines(self, chunk_size: int = None) -> Iterator[str]:
+        for text in self.iter_text(chunk_size):
+            lines = _parse_lines_from_text(text)
+            for line in lines:
+                yield line
+
+    def _close_stream(self) -> None:
+        self.is_stream_consumed = True
+        self.close()
+
+    def iter_raw(self, chunk_size: int = None) -> Iterator[bytes]:
+        """Iterate over the raw response bytes
+        """
+        self._validate_streaming_access()
+        stream_download = self._internal_response.stream_download(None)
+        for raw_bytes in stream_download:
+            self._num_bytes_downloaded += len(raw_bytes)
+            yield raw_bytes
+
+        self._close_stream()
+
+
+class AsyncHttpResponse(_HttpResponseBase):
+
+    @property
+    def content(self) -> bytes:
```

Comment:
Seems like this property could live on `_HttpResponseBase`?

Reply:
yep good point. it used to be different across sync and async, but no longer needed. Not going to update here though (it's fine to keep it like this), but will update in my core pr. thanks @annatisch !